wordpress-mobile/WordPress-Android
# Only include gb-mobile in vanilla and gutenberg flavors

Introduces a new build flavor (`gutenberg`) dedicated to include the gutenberg-mobile development flow. `vanilla` is also including the gutenberg-mobile stack as before.

`wasabi`, the typical development flavor used by the majority of the team's Android devs is only including a mocked version of the GutenbergEditorFragment and has no need to pull any submodules or install the RN tool stack.

Note: the Gutenberg-mobile editor is still hardcoded so, it will be used on vanilla and gutenberg flavors but not on zalpha and wasabi.

### To test A:
Build the app for the `wasabiDebug` build without having the gutenberg-mobile submodule pulled. Verify that wpandroid builds and runs fine otherwise.

## Before testing the rest of the cases below:
Follow the `Before testing` instructions from #8522 to setup the local gutenberg-mobile RN app.

### To test B:
Build the app for the `vanillaRelease` build and verify that it is producing the release build with the gutenberg-mobile app embedded and is able to open a Gutenberg post without the RN Metro server running.

### To test C:
Follow the instruction in Test case 1 from #8522 but use the `gutenbergDebug` build instead of `wasabiDebug` and verify that Gutenberg posts can be loaded and the gutenberg-mobile RN app is fetched from the locally running Metro server.
